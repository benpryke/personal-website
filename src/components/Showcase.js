import React from 'react';

import Banner from './Banner';

/**
 * A showcase of 1-3 showpieces, each represented via the Showpiece component
 */
export default class Showcase extends React.Component {
  render() {
    const { title, children, ...rest } = this.props;

    return (
      <Banner className='showcase' {...rest}>
        <h2>{title}</h2>
        <div className='showpieces'>
          {children}
        </div>
      </Banner>
    );
  }
}
