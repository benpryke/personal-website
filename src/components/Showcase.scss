.showcase {
  background: $background-colour;
  padding-bottom: 80px;
}

.showpieces {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  align-items: center;
  justify-content: center;

  a {
    color: rgb(43, 43, 43);
    text-decoration: none;
  }
}
