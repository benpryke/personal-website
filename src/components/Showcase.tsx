import React from 'react';

import Banner, { BannerPassthroughProps } from './Banner';

export interface ShowcaseProps extends BannerPassthroughProps {
  /** Heading */
  title: string,
  /** 1-3 showpieces */
  children: React.ReactNode;
}

/**
 * A showcase of 1-3 showpieces, each represented via the Showpiece component
 */
export default class Showcase extends React.Component<ShowcaseProps> {
  render() {
    const { title, children, ...rest } = this.props;

    return (
      <Banner className='showcase' {...rest}>
        <h2>{title}</h2>
        <div className='showpieces'>
          {children}
        </div>
      </Banner>
    );
  }
}
