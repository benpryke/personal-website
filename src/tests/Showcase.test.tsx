import React from "react";
import { render } from "@testing-library/react";
import pretty from "pretty";

import Showcase, { ShowcaseProps } from "../components/Showcase";

const props: ShowcaseProps = {
  title: "title",
  children: <div></div>,
};

describe("Showcase", () => {
  it("renders without crashing", () => {
    render(<Showcase {...props} />);
  });

  it("should match the snapshot", () => {
    const { container } = render(<Showcase {...props} />);
    expect(pretty(container.innerHTML)).toMatchSnapshot();
  });
});
