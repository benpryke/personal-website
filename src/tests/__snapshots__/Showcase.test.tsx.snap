// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Showcase > should match the snapshot 1`] = `
"<section class="banner showcase faded-out">
  <h2>title</h2>
  <div class="showpieces">
    <div></div>
  </div>
</section>"
`;
